File: SPRING/igest.server.model/src/main/java/com/yupanastudio/igest/server/principal/dml/Tipo.java
/*
 * Document   : Tipo
 * Author     : Franklin Barzola Olivares <frankoantonio@gmail.com>
 */
package com.yupanastudio.igest.server.principal.dml;

import org.springframework.data.annotation.Id;

public class Tipo {

    @Id
    private Integer idTipo;
    private String nomTipo;
    private String nomCorto;
    private String codOficial1;
    private String codOficial2;
    private String valor1;
    private String valor2;
    private String clave;
    private String grupo;
    private int orden;
    private Boolean activo;
    private String idSistema;
    private String descripcion;

    public Integer getIdTipo() {
        return idTipo;
    }

    public void setIdTipo(Integer idTipo) {
        this.idTipo = idTipo;
    }

    public String getNomTipo() {
        return nomTipo;
    }

    public void setNomTipo(String nomTipo) {
        this.nomTipo = nomTipo;
    }

    public String getNomCorto() {
        return nomCorto;
    }

    public void setNomCorto(String nomCorto) {
        this.nomCorto = nomCorto;
    }

    public String getCodOficial1() {
        return codOficial1;
    }

    public void setCodOficial1(String codOficial1) {
        this.codOficial1 = codOficial1;
    }

    public String getCodOficial2() {
        return codOficial2;
    }

    public void setCodOficial2(String codOficial2) {
        this.codOficial2 = codOficial2;
    }

    public String getValor1() {
        return valor1;
    }

    public void setValor1(String valor1) {
        this.valor1 = valor1;
    }

    public String getValor2() {
        return valor2;
    }

    public void setValor2(String valor2) {
        this.valor2 = valor2;
    }

    public String getClave() {
        return clave;
    }

    public void setClave(String clave) {
        this.clave = clave;
    }

    public String getGrupo() {
        return grupo;
    }

    public void setGrupo(String grupo) {
        this.grupo = grupo;
    }

    public int getOrden() {
        return orden;
    }

    public void setOrden(int orden) {
        this.orden = orden;
    }

    public Boolean getActivo() {
        return activo;
    }

    public void setActivo(Boolean activo) {
        this.activo = activo;
    }

    public String getIdSistema() {
        return idSistema;
    }

    public void setIdSistema(String idSistema) {
        this.idSistema = idSistema;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

}

File: FLUTTER/igest_app/lib/models/principal/tipo.dart
class Tipo {
  int idTipo;
  String nomTipo;
  String nomCorto;
  String codOficial1;
  String codOficial2;
  String codOficial3;
  String valor1;
  String valor2;
  String clave;
  String grupo;
  int orden;
  bool activo;
  String idSistema;
  String descripcion;

  Tipo({
    required this.idTipo,
    required this.nomTipo,
    required this.nomCorto,
    required this.codOficial1,
    required this.codOficial2,
    required this.codOficial3,
    required this.valor1,
    required this.valor2,
    required this.clave,
    required this.grupo,
    required this.orden,
    required this.activo,
    required this.idSistema,
    required this.descripcion,
  });

  factory Tipo.fromJson(Map<String, dynamic> json) {
    return Tipo(
      idTipo: json['id_tipo'],
      nomTipo: json['nom_tipo'],
      nomCorto: json['nom_corto'],
      codOficial1: json['cod_oficial1'],
      codOficial2: json['cod_oficial2'],
      codOficial3: json['cod_oficial3'],
      valor1: json['valor1'],
      valor2: json['valor2'],
      clave: json['clave'],
      grupo: json['grupo'],
      orden: json['orden'],
      activo: json['activo'],
      idSistema: json['id_sistema'],
      descripcion: json['descripcion'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_tipo': idTipo,
      'nom_tipo': nomTipo,
      'nom_corto': nomCorto,
      'cod_oficial1': codOficial1,
      'cod_oficial2': codOficial2,
      'cod_oficial3': codOficial3,
      'valor1': valor1,
      'valor2': valor2,
      'clave': clave,
      'grupo': grupo,
      'orden': orden,
      'activo': activo,
      'id_sistema': idSistema,
      'descripcion': descripcion,
    };
  }
}
